include/rocksdb/utilities/object_registry.h

```diff
@@ -38,6 +39,23 @@ using FactoryFunc =
 // library
 using RegistrarFunc = std::function<int(ObjectLibrary&, const std::string&)>;
 
+using PluginFunc = std::function<int(Plugin*, size_t, std::string*)>;
+
+struct Plugin {
+  // The version of this structure.  If the structure is changed, the version
+  // should be incremented
+  static constexpr int kVersion() { return 1; }
```

Comment:
This is unused. What is it doing?

Reply:
Originally, I had a few versions in this struct:
- The RocksDBVersion of the system/tool (hence the earlier changes to RocksDBVersion).  RocksDB would pass this value into the Plugin so it could determine that it supported the RocksDB version.  If the PluginFunc did not like the RocksDB version, it would return an error
- The RocksDB version of the Plugin.  This would be filled in by the plugin to say what version it was compiled against.  RocksDB would then check that the version was "okay" with it and not add the plugin if it was not
- The version of the Plugin struct.  This version was to be used by the plugin to state what fields it was filling in and aware of.  The the Plugin struct changed, the version of the struct would need to change so that the plugin would know.